demo/src/formValidator.js

```diff
@@ -37,6 +37,7 @@ export default form({
         },
         minLength({minLength: 4, message: 'Must have at least 4 characters'}),
         usernameIsAvailable,
+        // TODO: Is this actually doing anything?
```

Comment:
Sort of.  If the username is found to be valid, then the `every` validator will reach this point and the effect is adding a `successMessage` onto the validation result.

It's pretty hacky, actually.  Perhaps the `usernameIsAvailable` validator should always add a `successMessage` prop onto the validation result and the rendering would only consume that message if the field is valid.